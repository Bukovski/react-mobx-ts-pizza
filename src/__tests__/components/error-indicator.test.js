import React from 'react';
import { screen, render } from "@testing-library/react";
import { ErrorFallback, ErrorIndicator } from "../../components/error-indicator.component";


describe("ErrorIndicator Component", () => {
	it("ErrorIndicator should render", () => {
		const { container } = render(
			<ErrorIndicator>
				{ "Error text" }
			</ErrorIndicator>
		);
		
		expect(screen.getByText("Error text")).toBeInTheDocument();
		expect(container.firstChild).toHaveClass("error");
		
		expect(container.firstChild).toMatchSnapshot()
	})
	
	it("ErrorFallback should render", () => {
		const mockErrorFn = jest.fn();
		const saveConsoleError = console.error;
		
		console.error = mockErrorFn;
		
		render(<ErrorFallback />);
		
		expect(screen.getByText("Что-то пошло не так, но мы скоро все починим")).toBeInTheDocument();
		
		expect(mockErrorFn).toBeCalled();
		
		console.error = saveConsoleError;
	})
	
});
