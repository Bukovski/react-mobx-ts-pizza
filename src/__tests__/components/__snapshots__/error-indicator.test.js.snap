// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIndicator Component ErrorIndicator should render 1`] = `
<div
  class="error"
>
  <h3
    class="error--header"
  >
    Возникла ошибка
  </h3>
  <img
    alt="error-alert"
    class="error--image"
    src="error-img.png"
    title="error"
  />
  <p
    class="error--message"
  >
    Error text
  </p>
</div>
`;
